File: app/src/main/java/com/zhang/daka/daka/adapter/MenuAdapter.java
package com.zhang.daka.daka.adapter;

import android.support.annotation.Nullable;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.zhang.daka.R;

import java.util.List;

/**
 * Created by zhangyuncai on 2019/11/9.
 */
public class MenuAdapter extends BaseQuickAdapter<String, BaseViewHolder> {
    public MenuAdapter(@Nullable List<String> data) {
        super(R.layout.item_menu_adapter, data);
    }

    @Override
    protected void convert(BaseViewHolder helper, String item) {
        helper.setIsRecyclable(false);
        helper.setText(R.id.tv_menu_content, item);
    }
}

File: app/src/main/java/com/zhang/daka/ui/adapter/MenuAdapter.kt
package com.zhang.daka.ui.adapter

import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.zhang.daka.R
import com.zhang.daka.model.MenuModel

class MenuAdapter(list:ArrayList<MenuModel>):
        BaseQuickAdapter<MenuModel,BaseViewHolder>(R.layout.item_menu_adapter,list){
    override fun convert(helper: BaseViewHolder, item: MenuModel) {
        helper.setImageResource(R.id.iv_menu_icon,item.iconRes)
        helper.setText(R.id.tv_menu_name,item.menuName)
    }

}